torch/csrc/jit/runtime/register_ops_utils.h

```diff
@@ -652,6 +652,48 @@ void listSetItem(Stack* stack);
 //-----------------------------------------------------------------------------
 //-----------------------------------------------------------------------------
 //-----------------------------------------------------------------------------
+#define DEFINE_UNARY_COMPLEX_OP(aten_op, op, result)                      \
+  OperatorGenerator(                                                      \
+      TORCH_SELECTIVE_SCHEMA(#aten_op ".complex(complex a) -> " #result), \
+      [](Stack* stack) {                                                  \
+        c10::complex<double> a;                                           \
+        pop(stack, a);                                                    \
+        push(stack, op);                                                  \
+      },                                                                  \
+      aliasAnalysisFromSchema())
+
+#define DEFINE_UNARY_OP_WITH_COMPLEX_CAST(                                \
```

Comment:
Can you add some comments explaining what this is for and why it is needed? It seems to me that it only exists to make sure that `abs` has a float result.

Reply:
yeah it seemed a waste to use this everywhere since most unary ops output complex valued result. Only a few ops return real valued result. But good idea-- I'll add a comment!